tests/IntegrationTest.php

```diff
@@ -173,4 +174,44 @@ public function it_will_cache_file_responses()
 
         $this->assertSameResponse($firstResponse, $secondResponse);
     }
+
+    /** @test */
+    public function it_wont_cache_if_lifetime_is_0()
+    {
+        $this->app['config']->set('responsecache.cache_lifetime_in_minutes', 0);
+
+        $firstResponse = $this->call('get', '/');
+        $secondResponse = $this->call('get', '/');
+
+        $this->assertRegularResponse($firstResponse);
+        $this->assertRegularResponse($secondResponse);
+    }
+
+    /** @test */
+    public function it_will_cache_response_for_given_lifetime_which_is_defined_as_middleware_parameter()
+    {
+        // Set default lifetime as 0 to check if it will cache for given lifetime
+        $this->app['config']->set('responsecache.cache_lifetime_in_minutes', 0);
+
+        $firstResponse = $this->call('get', '/cache-for-given-lifetime');
+        $secondResponse = $this->call('get', '/cache-for-given-lifetime');
+
+        $this->assertRegularResponse($firstResponse);
+        $this->assertCachedResponse($secondResponse);
+    }
+
+    /** @test */
+    public function it_will_reproduce_cache_if_given_lifetime_is_expired()
+    {
+        // Set default lifetime as 0 to disable middleware that is already pushed to Kernel
+        $this->app['config']->set('responsecache.cache_lifetime_in_minutes', 0);
+
+        Carbon::setTestNow(Carbon::now()->subMinutes(6));
+        $firstResponse = $this->call('get', '/cache-for-given-lifetime');
```

Comment:
Can you call it for a second time here and assert that it now has cached the result?

Reply:
Done 👍